src/main/java/cart/controller/ExceptionItemControllerAdvice.java

```diff
@@ -0,0 +1,59 @@
+package cart.controller;
+
+import cart.controller.dto.ExceptionResponse;
+import cart.exception.NotFoundResultException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+// TODO: 2023/04/26 text/html을 기대할 떄와, application/json을 기대할 떄 똑같이 Json으로 예외 결과를 응답하는게 맞을까?
+@ControllerAdvice
+public class ExceptionItemControllerAdvice {
+
+    private static final Logger log = LoggerFactory.getLogger(ExceptionItemControllerAdvice.class);
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ExceptionResponse> handleMethodArgumentNotValidException(MethodArgumentNotValidException exception) {
+        log.info(exception.getMessage());
+        HttpStatus badRequest = HttpStatus.BAD_REQUEST;
+        ExceptionResponse exceptionResponse = new ExceptionResponse(badRequest.value(), badRequest.getReasonPhrase(), exception.getMessage());
+        return ResponseEntity.status(badRequest)
+                             .contentType(MediaType.APPLICATION_JSON)
+                             .body(exceptionResponse);
+    }
+
+    @ExceptionHandler(NotFoundResultException.class)
+    public ResponseEntity<ExceptionResponse> handleNotFoundResultException(NotFoundResultException exception) {
+        log.info(exception.getMessage());
+        HttpStatus badRequest = HttpStatus.BAD_REQUEST;
+        ExceptionResponse exceptionResponse = new ExceptionResponse(badRequest.value(), badRequest.getReasonPhrase(), exception.getMessage());
+        return ResponseEntity.status(badRequest)
+                             .contentType(MediaType.APPLICATION_JSON)
+                             .body(exceptionResponse);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<ExceptionResponse> handlerIllegalArgumentException(IllegalArgumentException exception) {
+        log.info(exception.getMessage());
+        HttpStatus badRequest = HttpStatus.BAD_REQUEST;
+        ExceptionResponse exceptionResponse = new ExceptionResponse(badRequest.value(), badRequest.getReasonPhrase(), exception.getMessage());
+        return ResponseEntity.status(badRequest)
```

Comment:
```suggestion
        return ResponseEntity.badRequest()
```

이러면 위에서 HttpStatus 변수를 안 만들어도 되겠네요 😎

Reply:
완벽히 해결되지는 않은 상태입니다…!
다만, test/html을 기대하는 경우 현재 상태에서는 모두 application/json으로 예외를 전달하기에 예외가 발생한 경우 아래와 같이 예외 결과가 보이게 됩니다.

![image](https://user-images.githubusercontent.com/63184334/235467055-27c8aaef-db2c-4ee8-ae78-3c9c8fbaed05.png)

현재 예외 페이지가 따로 존재하는 것이 아니긴 하지만 이는 클라이언트 측에서 원하는 예외 발생의 응답은 아니라고 생각하기에 test/html로 반환하는 ExceptionControllerAdvice가 필요하지 않을까 생각합니다.
이에 대한 춘식의 의견도 알려주시면 감사하겠습니다!